### PR TITLE
Communicate tenant info in changes

### PR DESCRIPTION
**Pull request by mem**
When then tenant remote information changes, the API will communicate
this to the probes in the changes stream.

The GetTenant service is still needed, because there might be changes
happening outside the API (for example, the user deleted a token), so
the probes still need to be able to query the tenant information on
their own.

Changing the name of the protocol buffer message is backwards
compatible.

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

### CODE REVIEW

**Comment by mem** [comment]
I verified this is working with the existing API

**Comment by mem** [comment]
> I verified this is working with the existing API

Correction: I verified this before the very last change.

File: internal/pusher/pusher.go [hunk 1]
```diff
@@ -208,37 +222,51 @@ func (p *Publisher) getClient(ctx context.Context, tenantId int64, newClient boo
...
 		Metrics: mClient,
 		Events:  eClient,
 	}
+
 	p.clientsMutex.Lock()
-	p.clients[tenantId] = clients
+	p.clients[tenant.Id] = clients
 	p.clientsMutex.Unlock()
 
+	p.logger.Debug().Interface("tenant", tenant).Msg("updated client")
```
Comment (electron0zero) [inline internal/pusher/pusher.go:268]: let's log tenantId as well

**Review by electron0zero — APPROVED** [review]
few nits, rest LGTM :+1:

**Pull request merged**